dts/x86/ia32.dtsi

```diff
@@ -28,18 +28,17 @@
 	};
 
 
-	flash0: flash@1000 {
+	flash0: flash@100000 {
```

Comment:
This all seems fine, low memory is a booby trap, in particular the 640k-1M region which has the MMIO mappings for legacy devices.  Note that real hardware does have (and require) this memory though, so it's likely we'll have to come back here at some point and put in some kind of mapping/allocation layer to allow us to get to it.  But loading the binary at 1M is IMHO absolutely the right choice.